### PR TITLE
Misc changes made during Update work

### PR DESCRIPTION
**Pull request by dandavison**
- Rename `update/intercept` feature to `/update/client_interceptor`

- Add a feature demonstrating that activities can be run in a Signal handler, analogous to the `update/activities` feature that demonstrates this for Update handlers.

### CODE REVIEW

File: features/signal/activities/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Run activities from within a Workflow Signal
+
+Workflows can run activities from within a Signal handler.
+
+# Detailed spec
+
+Workflow Signal handlers are like workflow code in that they can invoke and wait on activities.
+This test invokes a number of activities within the Signal and blocks on them, returning from the Signal when they have all completed.
```
Comment (dandavison) [inline features/signal/activities/README.md:8]: Is this a feature worth having? I created it to test my own understanding, and I couldn't think of a reason why this feature should exist for Update and not Signal and Query (which someone can add later).

Comment (cretz) [reply features/signal/activities/README.md:8]: Sure, worth having IMO. Anything that is a clear callout to something we need to support can be a feature.

File: features/update/client_interceptor/feature.go
Comment (cretz) [inline features/update/client_interceptor/feature.go:1]: :+1: On the name change. Not sure why the intercept feature didn't include workflow inbound interceptors.

**Review by cretz — APPROVED** [review]

**Comment by dandavison** [comment]
Thanks @cretz.

**Pull request merged**